opm/autodiff/createGlobalCellArray.hpp

```diff
@@ -34,7 +34,7 @@ void createGlobalCellArray(const Grid &grid, std::vector<int>& dest)
     dest.resize(numCells);
     const auto& globalCell = Opm::UgGridHelpers::globalCell(grid);
     std::vector<int> compressedToCartesianIdx(numCells);
-    for (unsigned cellIdx = 0; cellIdx < numCells; ++cellIdx) {
+    for (int cellIdx = 0; cellIdx < numCells; ++cellIdx) {
```

Comment:
> May I suggest a somewhat more substantial rewrite of this function?

I will adapt it to use compressedToCartesian, but for now I'd just like to not get annoyed by the warnings.